lib/compress/zstd_compress_internal.h

```diff
@@ -1198,6 +1198,9 @@ size_t ZSTD_referenceExternalSequences(ZSTD_CCtx* cctx, rawSeq* seq, size_t nbSe
 
 /** ZSTD_cycleLog() :
  *  condition for correct operation : hashLog > 1 */
+#if defined(ZSTD_PROGRAMS_LINK_SHARED)
```

Comment:
This could instead be some conditionally-defined `ZSTDLIB_CLI_DLL_API` (or whatever) in lib/zstd.h.  I don't feel strongly about it; the number of instances needed will probably remain very small.

Reply:
We could probably remove `ZSTD_cycleLog()` dependency. It's trivial to copy.